packages/gestalt/src/boxTransforms.js

```diff
@@ -0,0 +1,648 @@
+          value.column ? columnRange(value.column) : identity(),
+          typeof value.display !== 'undefined'
+            ? displayFunctor(value.display)
+            : identity(),
+        ])
+      )
+    );
+  },
+  sm: (value: ?ResponsiveProps): Style => {
+    if (!value) {
+      return identity();
+    }
+    return mapClassName(c => styles[c])(
+      prefix('sm')(
+        concat([
+          value.column ? columnRange(value.column) : identity(),
+          typeof value.display !== 'undefined'
+            ? displayFunctor(value.display)
+            : identity(),
+        ])
+      )
+    );
+  },
+  md: (value: ?ResponsiveProps): Style => {
+    if (!value) {
+      return identity();
+    }
+    return mapClassName(c => styles[c])(
+      prefix('md')(
+        concat([
+          value.column ? columnRange(value.column) : identity(),
+          typeof value.display !== 'undefined'
+            ? displayFunctor(value.display)
+            : identity(),
+        ])
+      )
+    );
+  },
+  lg: (value: ?ResponsiveProps): Style => {
+    if (!value) {
+      return identity();
+    }
+    return mapClassName(c => styles[c])(
+      prefix('lg')(
+        concat([
+          value.column ? columnRange(value.column) : identity(),
+          typeof value.display !== 'undefined'
+            ? displayFunctor(value.display)
+            : identity(),
+        ])
+      )
+    );
+  },
+
+  display,
+  column,
+  direction,
+
+  smDisplay,
+  smColumn,
+  smDirection,
+
+  mdDisplay,
+  mdColumn,
+  mdDirection,
+
+  lgDisplay,
+  lgColumn,
+  lgDirection,
+
+  alignContent,
+  alignItems,
+  alignSelf,
+  bottom,
+  borderSize,
+  color,
+  fit,
+  flex,
+  height,
+  justifyContent,
+  left,
+
+  margin,
+  marginTop,
+  marginRight,
+  marginBottom,
+  marginLeft,
+  marginStart,
+  marginEnd,
+
+  smMargin,
+  smMarginTop,
+  smMarginRight,
+  smMarginBottom,
+  smMarginLeft,
+  smMarginStart,
+  smMarginEnd,
+
+  mdMargin,
+  mdMarginTop,
+  mdMarginRight,
+  mdMarginBottom,
+  mdMarginLeft,
+  mdMarginStart,
+  mdMarginEnd,
+
+  lgMargin,
+  lgMarginTop,
+  lgMarginRight,
+  lgMarginBottom,
+  lgMarginLeft,
+  lgMarginStart,
+  lgMarginEnd,
+
+  maxHeight,
+  maxWidth,
+  minHeight,
+  minWidth,
+  opacity,
+  overflow,
+
+  padding,
+  paddingX,
+  paddingY,
+  smPadding,
+  smPaddingX,
+  smPaddingY,
+  mdPadding,
+  mdPaddingX,
+  mdPaddingY,
+  lgPadding,
+  lgPaddingX,
+  lgPaddingY,
+
+  position,
+  right,
+  rounding,
+  top,
+  userSelect,
+  width,
+  wrap,
+  dangerouslySetInlineStyle,
+  zIndex,
+};
+
+/*
+
+# Style Builder
+
+This is where it all comes together. This function takes the base styles for the component,
+the component's props, and any disallowed props. It outputs the passthrough props (after
+removing disallowed and given props), as well as an object with the combined classNames
+and styles from the given props.
+
+*/
+
+// $FlowExpectedError[unclear-type]
+export function buildStyles<T: Object>(
```

Comment:
we know that the generic type will be an object, but we don't know anything else about it